sources/packages/backend/apps/api/src/institution-integration/ier-integration/ier12-file.service.ts

```diff
@@ -0,0 +1,172 @@
+      )) {
+        uploadResult.push(ierUploadResult);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for IER 12 Request: ${error}`,
+      );
+      throw error;
+    }
+    return uploadResult;
+  }
+
+  /**
+   * Upload the content in SFTP server location.
+   * @param institutionCode Institution code for the file generated.
+   * @param fileRecords Total records with institutionCode.
+   * @returns Updated records count with filepath.
+   */
+  async uploadIER12Content(
+    institutionCode: string,
+    fileRecords: Record<string, IER12Record[]>,
+  ): Promise<IER12UploadResult> {
+    try {
+      // Create the Request content for the IER 12 file by populating the content.
+      const fileContent = this.ierIntegrationService.createIER12FileContent(
+        fileRecords[institutionCode],
+      );
+      // Create the request filename with the file path for the each and every institutionCode.
+      const fileInfo = this.createRequestFileName(institutionCode);
+      this.logger.log("Uploading content...");
+      await this.ierIntegrationService.uploadContent(
+        fileContent,
+        fileInfo.filePath,
+      );
+      this.logger.log("Content uploaded.");
+      return {
+        generatedFile: fileInfo.filePath,
+        uploadedRecords: fileContent.length,
+      };
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for IER 12 file: ${error}`,
+      );
+      throw error;
+    }
+  }
+
+  /**
+   * Create file name IER 12 records file.
+   * @param institutionCode institutionCode to be a part of the filename.
+   * @returns Full file path of the file to be saved on the SFTP.
+   */
+  createRequestFileName(institutionCode: string): {
+    fileName: string;
+    filePath: string;
+  } {
+    const timestamp = getFileNameAsCurrentTimestamp();
+    const fileName = `IER_012_${timestamp}.txt`;
+    const filePath = `${this.InstitutionIntegrationConfig.ftpRequestFolder}\\${institutionCode}\\${fileName}`;
+    return {
+      fileName,
+      filePath,
+    };
+  }
+  private createIER12Record(pendingAssessment: StudentAssessment): IER12Record {
+    const application = pendingAssessment.application;
+    const student = application.student;
+    const user = student.user;
+    const sinValidation = student.sinValidation;
+    const offering = pendingAssessment.offering;
+    const educationProgram = offering.educationProgram;
+    return {
+      assessmentId: pendingAssessment.id,
+      applicationNumber: application.applicationNumber,
+      sin: sinValidation.sin,
+      studentLastName: user.lastName,
+      studentGivenName: user.firstName,
+      birthDate: student.birthDate,
+      programName: educationProgram.name,
+      programDescription: educationProgram.description,
+      credentialType: educationProgram.credentialType,
+      cipCode: parseFloat(educationProgram.cipCode) * 10000,
```

Comment:
I believe that this types of transformation belong to the getFixedString method while generating the file content.
Why are we multiplying by 10000 here?

Reply:
just to remove the decimal point multiplies with 10000 for simplicity